### PR TITLE
feature: #37 | Update the Response Structure with the new learnings

### PR DESCRIPTION
**Pull request by astha-singh-dev**

### CODE REVIEW

File: src/main/java/dev/shivamnagpal/users/exceptions/handlers/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -21,7 +21,7 @@ public static void handle(Throwable throwable, HttpServerResponse httpServerResp
         log.error(throwable.getMessage(), throwable);
         if (throwable instanceof RestException restException) {
             ResponseWrapper<Object> failureResponseWrapper = ResponseWrapper
-                    .failure(restException.getErrorResponses(), restException.getPayload());
+                    .failure(restException.getErrorResponses());
```
Comment (ShivamNagpal) [inline src/main/java/dev/shivamnagpal/users/exceptions/handlers/GlobalExceptionHandler.java]: Why did you remove the restException.getPayload() from here?
It is needed for one case where we send the OtpToken if the user tries to login and his account isn't verified

Comment (ShivamNagpal) [reply src/main/java/dev/shivamnagpal/users/exceptions/handlers/GlobalExceptionHandler.java]: To solve the issue
Create another failure() method overload and it should accept List<ErrorResponses> and T data

**Review by ShivamNagpal — CHANGES_REQUESTED** [review]

File: src/main/java/dev/shivamnagpal/users/exceptions/handlers/GlobalExceptionHandler.java (hunk 1 above)
Comment (astha-singh-dev) [reply src/main/java/dev/shivamnagpal/users/exceptions/handlers/GlobalExceptionHandler.java]: fixed

File: src/main/java/dev/shivamnagpal/users/dtos/response/wrapper/ResponseWrapper.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                .build();
+        return new ResponseWrapper<>(data, pagination, null, metaResponse);
+    }
+
+    public static <T> ResponseWrapper<T> failure(List<ErrorResponse> errorResponses) {
+        MetaResponse metaResponse = MetaResponse.builder()
+                .status(ResponseStatus.FAILURE)
+                .timestamp(Instant.now().toEpochMilli())
+                .build();
+        return new ResponseWrapper<>(null, null, errorResponses, metaResponse);
```
Comment (ShivamNagpal) [inline src/main/java/dev/shivamnagpal/users/dtos/response/wrapper/ResponseWrapper.java]: Just call the bigger overload method here
`return failure(errorResponses, null)`

**Review by ShivamNagpal — CHANGES_REQUESTED** [review]

File: src/main/java/dev/shivamnagpal/users/dtos/response/wrapper/ResponseWrapper.java (hunk 2 above)
Comment (astha-singh-dev) [reply src/main/java/dev/shivamnagpal/users/dtos/response/wrapper/ResponseWrapper.java]: fixed

**Review by ShivamNagpal — APPROVED** [review]
